general/documentation/contributing.md

```diff
@@ -31,6 +33,77 @@ Anyone can get involved with, and add to, these resources - you don't have to be
 
+### Migrating legacy docs
+
+We are currently migrating documentation from our [legacy documentation site](https://docs.moodle.org/dev), and one of the best ways to contribute with documentation is to help with this migration process.
+
+We have created a number of tools to make this a little easier and to automate as much as possible. These are mostly focused around our migration assistant. You can find its full documentation by running:
+
+```console
+yarn migrate --help
+```
+
+#### What we automate
+
+The migration assistant will:
+
+- try to guess the:
+  - page title
+  - relevant tags
+- create the directory structure to the new markdown file
+- fetch any images included in the original page
+- convert:
+  - WikiText tables to either Markdown tables, or HTML tables as appropriate
+  - page headings
+  - lists
+  - common font formatting like bold, and italics
+  - code blocks
+  - some common `*Note*` syntaxes to use an [Admonition](https://docusaurus.io/docs/next/markdown-features/admonitions)
+  - InterWiki links to fully-qualified links
+  - Regular links to markdown links
+- run markdownlint in fix mode to automatically fix any common, fixable issues
+- update the links for any internal documentation which has already been migrated
+- add the migrated page to our [list of migrated pages](https://github.com/moodle/devdocs/main/blob/data/migratedPages.yml)
+- update any links to the migrated page
+
+#### Using the assistant
+
+In most circumstances you should not need to pass any special options, and you will just need to pass the name of the document you are migrating, and the path to the location you'd like to migrate it to, for example:
+
+```console title="Migrate the 'Moodle and PHP' documentation to a new policies page"
+yarn migrate [Name or URL of legacy document] [/path/to/new/document]
+```
+
+:::tip
+
+You can either pass the full URL to the legacy document on the legacy devdocs site, or the name of the legacy document is the name within the URL, for example, in the following URL:
+
+```
+https://docs.moodle.org/dev/Moodle_and_PHP
+```
+
+The page name is `"Moodle_and_PHP"`.
+
+:::
+
+When choosing a location to migrate the document to, first ask whether the page contains information which relates to a particular version of Moodle - this may be a guide for a subsystem, or plugintype for example. If so, then this _must_ go into the `docs` folder. Most other content should go into the `general` folder.
+
+:::info
+
+The [structure of the documentation](./structure.md#general-structure) describes these two locations in further detail.
+
+:::
+
+```console title="Migrate the 'Moodle and PHP' documentation to a new policies page"
+yarn migrate Moodle_and_PHP ./general/development/policies/php.md
+```
+
+:::note
+
+The migration assistant is not perfect and is not aware of all WikiText features. Additionally if the formatting is not correct in the WikiText, some aspects may become incorrectly formatted during the conversion.
```

Comment:
It would be nice to add some extra comments, related to this, to explain that it's important to review the page, update the content and improve the styling taking advantage of the docusaurus features.